File: codechefjan20bCHFDORA.c
#include <stdio.h>
int main()
{int t;
int a[100005][100005];
    scanf("%d",&t);
    while(t--)
    {int m,n;
    scanf("%d %d",&m,&n);
    int total=m*n;
    for (int i=0;i<m;i++)
    {for(int j=0;j<n;j++)
    scanf("%d",&a[i][j]);}
    for(int i=1;i<m;i++)
    {
    for(int j=1;j<n;j++)
    {int i1=1;
	while(i>=i1 && i+i1<m && j>=i1 && j+i1<n)
    {if (a[i][j-i1]==a[i][j+i1] && a[i-i1][j]==a[i+i1][j])
    {total+=1;i1+=1;}
    else
    break;}}
    }
    printf("%d\n",total);
    }
return 0;
    }

File: codechefjan20bCHFDORA.py
for _ in range(int(input())):
    m,n=map(int,input().split())
    li=[[0 for i in range(m)] for j in range(n)]
    for i in range (m):
        li[i]=list(map(int,input().split()))
    total=n*m
    for i in range(1,m):
        for j in range (1,n):
            i1=1
            while(i-i1>=0 and i+i1<m and j-i1>=0 and j+i1<n):
                if (li[i-i1][j]==li[i+i1][j] and li[i][j-i1]==li[i][j+i1]):
                    total+=1
                    i1+=1
                else:
                    break
    print(total)
